reframe/utility/typecheck.py

```diff
@@ -0,0 +1,244 @@
+"""Dynamic recursive type checking of aggregate data structures.
+
+This module defines types for aggregate data structures, such as lists,
+dictionaries etc. that you can use with the ``isinstance`` builtin function to
+recursively check of all the elements of an aggregate data structure.
+Suppose you have a list of integers, suchs as ``[1, 2, 3]``, the following
+checks should hold:
+
+::
+    l = [1, 2, 3]
+    assert isinstance(l, List[int]) == True
+    assert isinstance(l, List[false]) == False
```

Comment:
Do you mean `assert isinstance(l, List[bool]) == False`

Reply:
Yes!